### PR TITLE
fix: `lerna diff` for repositories without commits

### PR DESCRIPTION
**Pull request by joscha**
`lerna diff` on a repository without any commits fails due to:
```console
$ lerna diff
Lerna v2.0.0-beta.7
Independent Versioning Mode
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Errored while running DiffCommand.initialize
Error: Command failed: git rev-list --max-parents=0 HEAD
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

    at checkExecSyncError (child_process.js:470:13)
    at Object.execSync (child_process.js:510:13)
    at Function.execSync (/usr/local/lib/node_modules/lerna/lib/ChildProcessUtilities.js:37:38)
    at Function.getFirstCommit (/usr/local/lib/node_modules/lerna/lib/GitUtilities.js:92:46)
    at DiffCommand.initialize (/usr/local/lib/node_modules/lerna/lib/commands/DiffCommand.js:63:130)
    at DiffCommand._attempt (/usr/local/lib/node_modules/lerna/lib/Command.js:148:21)
    at DiffCommand.runCommand (/usr/local/lib/node_modules/lerna/lib/Command.js:132:12)
    at DiffCommand.run (/usr/local/lib/node_modules/lerna/lib/Command.js:62:12)
    at Object.<anonymous> (/usr/local/lib/node_modules/lerna/bin/lerna.j%
```
currently.

### CODE REVIEW

File: src/GitUtilities.js [hunk 1]
```diff
@@ -73,4 +73,9 @@ export default class GitUtilities {
   static getCurrentBranch() {
     return ChildProcessUtilities.execSync("git symbolic-ref --short HEAD");
   }
+
+  @logger.logifySync
+  static hasCommit() {
+    return ChildProcessUtilities.execSync("git log");
```
Comment (thejameskyle) [inline src/GitUtilities.js]: Can you make this return a boolean instead: 

```js
!!ChildProcessUtilities.execSync("git log");
```

File: src/commands/DiffCommand.js [hunk 2]
```diff
@@ -38,4 +38,15 @@ export default class DiffCommand extends Command {
       }
     });
   }
+
+  runValidations() {
+    super.runValidations();
+    try {
+      GitUtilities.hasCommit();
```
Comment (thejameskyle) [inline src/commands/DiffCommand.js]: Oh never mind the previous comment, I see what you are doing.

Could you actually move this to `initialize()` and create a version of `hasCommit` that doesn't throw and just returns a boolean?

**Comment by thejameskyle** [comment]
Thanks for tackling this @joscha

**Comment by joscha** [comment]
@thejameskyle I made the changes, let me know if that works for you. I can also squash the commits in case you'd like that.

**Comment by hzoo** [comment]
We can squash in the merge now so it's ok

**Comment by thejameskyle** [comment]
Thanks!

**Comment by joscha** [comment]
Thank you for merging - one thing I thought with the errors we have now in quite a few places - would you be happy to integrate a PR that unifies and cleans up the errors using https://www.npmjs.com/package/define-error ?

**Comment by thejameskyle** [comment]
Yeah, errors and error handling are definitely places I want to be focusing on. I'm not sure what that package would do to help though.,

**Comment by joscha** [comment]
I think exporting the errors and making them importable at the right places would be the first step. The package itself only makes sure that there is a proper stack trace attached.

**Pull request merged**